Stratis.Bitcoin/Features/BlockStore/LoopSteps/DownloadBlockStep.cs

```diff
@@ -0,0 +1,176 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.BlockPulling;
+using Stratis.Bitcoin.BlockStore;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
+{
+    // Continuously download blocks until a stop condition is found.
+    // There are two operations:
+    //      Find blocks to download by asking them from the BlockPuller
+    //      Collecting downloaded blocks and persisting them as a batch to the BlockRepository
+
+    internal sealed class DownloadBlockStep : BlockStoreLoopStep
+    {
+        internal DownloadBlockStep(BlockStoreLoop blockStoreLoop)
+            : base(blockStoreLoop)
+        {
+        }
+
+        private int insertBlockSize = 0;
+        private int stallCount = 0;
+        private List<BlockPair> blockPairsToStore = new List<BlockPair>();
+
+        internal override async Task<BlockStoreLoopStepResult> Execute(ChainedBlock nextChainedBlock, CancellationToken cancellationToken, bool disposeMode)
```

Comment:
My take here is preserving the original code, to me it was more readable and then you don't need to preserve state between methods. 
You then wont need a dedicated DownloadBlockStepResult 

But you might have a point with breaking the code further.